pybamm/parameters/update_parameter_sets_doc.py

```diff
@@ -0,0 +1,102 @@
+"""
+Automatically update the docstring of parameter_sets.py
+"""
+
+from collections import defaultdict
+import os
+import re
+
+import pybtex
+
+import pybamm
+from pybamm.parameters import parameter_sets
+
+
+DOC_INTRO = """
+Parameter sets from papers. The 'citation' entry provides a reference to the appropriate
+paper in the file "pybamm/CITATIONS.txt". To see which parameter sets have been used in
+your simulation, add the line "pybamm.print_citations()" to your script."""
+
+
+def get_ps_dict():
```

Comment:
can you add a test for this function too?

Reply:
This function uses the `parameter_sets.py` module to generate chemistry-citation pairs. Every time someone adds or removes a new citation, the test will need to be updated. Is that okay?

Reply:
The test does not need to check the whole dictionary. For example it could simply test that "Marquis2019" (with 1 citation) and "Ecker2015" (with 2 citations) are both keys in the dict with the correct values